generate/homogenize_xml_field_format.py

```diff
@@ -0,0 +1,204 @@
+                                                         python_format, str(node.text)))
+                if len(T) == 0:
+                    T['field_nr'] = ['{:>5}'.format(field_number)]
+                    T['field_name'] = ['{:>20}'.format(node.tag)]
+                    T['format'] = ['{:>10}'.format(prd_data_class)]
+                    T['pyformat'] = ['{:>10}'.format(python_format)]
+                    T['example'] = ['{:>30}'.format(node.text)]
+                else:
+                    T.add_row((field_number, node.tag, prd_data_class, python_format, node.text))
+
+            field_index += 1
+
+    return T
+
+
+xml_formats = {}
+for instrument in ['NIRISS', 'FGS', 'NIRSpec']:
+
+    siaf = pysiaf.Siaf(instrument)
+
+    pre_delivery_dir = os.path.join(JWST_DELIVERY_DATA_ROOT, instrument)
+    if not os.path.isdir(pre_delivery_dir):
+        os.makedirs(pre_delivery_dir)
+
+    # write the SIAF files to disk
+    filenames = pysiaf.iando.write.write_jwst_siaf(siaf, basepath=pre_delivery_dir,
+                                                   file_format=['xml', 'xlsx'])  # , label='update'
+
+
+    if 1:
+        xml_formats[instrument] = read_xml_field_formats(instrument, filenames[0])
+
+    if 0:
+        pre_delivery_siaf = pysiaf.Siaf(instrument, basepath=pre_delivery_dir)
+
+        ref_siaf = pysiaf.Siaf(instrument)
+        compare.compare_siaf(pre_delivery_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6,
+                             tags={'reference': pysiaf.JWST_PRD_VERSION, 'comparison': 'pre_delivery'})
+        compare.compare_siaf(pre_delivery_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6,
+                             report_dir=pre_delivery_dir, tags={'reference': pysiaf.JWST_PRD_VERSION, 'comparison': 'pre_delivery'})
+
+        if instrument.lower() not in ['nirspec']:
+            # run some tests on the new SIAF
+            from pysiaf.tests import test_aperture
+            print('\nRunning aperture_transforms test for pre_delivery_siaf')
+            test_aperture.test_jwst_aperture_transforms([pre_delivery_siaf], verbose=False)#, threshold=0.1)
+            print('\nRunning aperture_vertices test for pre_delivery_siaf')
+            test_aperture.test_jwst_aperture_vertices([pre_delivery_siaf])
+        else:
+            from pysiaf.tests import test_nirspec
+
+            print('\nRunning regression test of pre_delivery_siaf against IDT test_data:')
+            test_nirspec.test_against_test_data(siaf=pre_delivery_siaf)
+
+            print('\nRunning nirspec_aperture_transforms test for pre_delivery_siaf')
+            test_nirspec.test_nirspec_aperture_transforms(siaf=pre_delivery_siaf, verbose=False)
+
+            print('\nRunning nirspec_slit_transforms test for pre_delivery_siaf')
+            test_nirspec.test_nirspec_slit_transformations(siaf=pre_delivery_siaf, verbose=False)
+
+show_field_formats = True
+
+if show_field_formats:
```

Comment:
Could this also be a separate function that can get called in the file? Just to keep things separate

Reply:
done